lib/codebuild-stack.ts

```diff
@@ -82,6 +83,14 @@ export class CodeBuildStack extends cdk.Stack {
         }
         const machineImage = new ec2.LookupMachineImage(machineImageProps);
 
+        const fleetServiceRole = new iam.Role(this, `FleetServiceRole-${toStackName(props.arch)}`, {
```

Comment:
We should include the operating system here as well:
```suggestion
        const fleetServiceRole = new iam.Role(this, `FleetServiceRole-${operatingSystem}-${toStackName(props.arch)}`, {
```

Can you also do the same for the `encryptionKey` while you're at it?